internal/common/windows_utils.bzl

```diff
@@ -0,0 +1,66 @@
+# Copyright 2017 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Windows Utils
+
+These functions help making rules work on Windows.
+"""
+
+def is_windows(ctx):
```

Comment:
there's already a couple other implementations of `is_windows` logic in  https://github.com/bazelbuild/rules_nodejs/blob/master/internal/node/node_repositories.bzl
https://github.com/bazelbuild/rules_nodejs/blob/master/third_party/github.com/bazelbuild/bazel-skylib/rules/private/copy_file_private.bzl

what do you think about reducing this ?

Reply:
Thanks for the suggestion!

The implementation in node_repositories.bzl is for repository rule only (requires `repository_ctx`), the `repository_ctx.os.name` API is not available in rule context (`ctx`). So we cannot use that.

The implementation in copy_file_private.bzl looks more reasonable. But it requires us to add `is_windows` attribute to every rule that wants to distinguish platforms and a macro to provide default value for that attribute. But essentially, it will give us the same result. Considering we are not going to do cross-platform build between Windows and Unix systems in near future, I think the current way is fine.